### ISSUE TITLE
Switch from Cloud Run to Compute Engine

### ISSUE THREAD

**Issue by smonn** [deptagency/algomart#20]
## Overview

As Cloud Run imposes some limitations that are proving hard to workaround, let's switch to Compute Engine (or some other non-serverless GCP option).

**Comment by kevlarr** [comment]
Before committing to the overhaul, I'd like to at least verify that using GCS as a backing store for the CMS isn't also an issue. That is, we should make sure that we can use Directus locally but point to GCS rather than local disk storage.

**Comment by brandonaaskov** [comment]
If we find that this is a pain, we can also lean on Google Cloud Storage for this stuff. Asking a customer to encode video for the previews isn't asking too much IMO, especially if they're already selling a source video asset that they likely edited/created/exported.

**Comment by kevlarr** [comment]
@brandonaaskov Are you suggesting that they only upload the smaller preview through Directus and only link to the real one in the asset URL or something?

**Comment by rtwall** [comment]
Hey, I just discovered this project. [GKE Autopilot](https://cloud.google.com/kubernetes-engine/docs/concepts/autopilot-overview) might be a reasonable option depending on the Cloud Run limitations you're having. Autopilot comes with some limitations as well.

There is an [enable_autopilot](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_cluster#enable_autopilot) argument on the Terraform cluster resource.

**Comment by kevlarr** [comment]
Welcome @rtwall! Thanks for the suggestion - any particular limitations with Autopilot that come to mind?

**Comment by kevlarr** [comment]
Awesome, thanks @rtwall! It's really helpful to get context like this *before* just diving in

**Comment by theothermattm** [comment]
I'm not sure that pursuing GCP specific options is the best idea if you're aiming for wide adoption in an open source project. 
 Though it might be heavy handed, I would consider just moving to a Kubernetes based setup which could be implemented on any cloud provider. @smonn and @juttayaya your thoughts.

**Comment by smonn** [comment]
> I'm not sure that pursuing GCP specific options is the best idea if you're aiming for wide adoption in an open source project. Though it might be heavy handed, I would consider just moving to a Kubernetes based setup which could be implemented on any cloud provider. @smonn and @juttayaya your thoughts.

@theothermattm Good point! We'll be going for Kubernetes to avoid vendor lock-in. The Terraform setup will be different no matter what, but as long as it can run in a Kubernetes environment I think we'll be in a good place.
